### DISCUSSION TITLE
Issue updating Docker image digest for custom manager configuration

### DISCUSSION THREAD

**Discussion by juusopiikkila** [renovatebot/renovate#28355] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I'm facing an issue with Renovate failing to update the digest value for a Docker image when using a custom manager configuration. Here are the details:

**Renovate Configuration:**
```json
{
  "customManagers": [
    {
      "customType": "regex",
      "datasourceTemplate": "docker",
      "fileMatch": ["<file-name>.yaml$"],
      "matchStrings": ["<image-reference-pattern>: (?<depName>.*?)@(?<currentValue>.*?)\\s"],
      "versioningTemplate": "docker"
    }
  ]
}
```

**File Content (<file-name>.yaml):**
```yaml
helm:
  values: |
    image:
      <image-reference-pattern>: ghcr.io/<org>/<image-name>@sha256:84ee5d7c24a7febfccda68b957d227e43f47c2c52a71bbf02133787456ade970
```

**Renovate Log:**
```
DEBUG: Found match at index 660 (branch="renovate/pin-dependencies")
{
  "packageFile": "<file-path>/<file-name>.yaml"
  "depName": "ghcr.io/<org>/<image-name>"
}

DEBUG: Digest is not updated (branch="renovate/pin-dependencies")
{
  "manager": "regex"
  "packageFile": "<file-path>/<file-name>.yaml"
  "expectedValue": "sha256:84ee5d7c24a7febfccda68b957d227e43f47c2c52a71bbf02133787456ade970"
}

WARN: Error updating branch: update failure (branch="renovate/pin-dependencies")
```

The issue is that Renovate is failing to update the digest value for the `ghcr.io/<org>/<image-name>` image, even though the latest digest is `sha256:0e121416c567e8574beeb37e0085197c2213824ffab799ce4f6678e48f7c07b5`. I've verified that Renovate can authenticate with the `ghcr.io` registry using the correct permissions.

Could you please help me understand why Renovate is unable to update the digest value in this case? Is there an issue with the custom manager configuration or the way the image reference is formatted in the file?

Any guidance or suggestions on how to resolve this issue would be greatly appreciated.

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Found match at index 660 (branch="renovate/pin-dependencies")
{
  "packageFile": "<file-path>/<file-name>.yaml"
  "depName": "ghcr.io/<org>/<image-name>"
}

DEBUG: Digest is not updated (branch="renovate/pin-dependencies")
{
  "manager": "regex"
  "packageFile": "<file-path>/<file-name>.yaml"
  "expectedValue": "sha256:84ee5d7c24a7febfccda68b957d227e43f47c2c52a71bbf02133787456ade970"
}

WARN: Error updating branch: update failure (branch="renovate/pin-dependencies")


```

</details>

**Comment by viceice** [comment]
you need to add current digest to match strings groups

**Reply by juusopiikkila** [reply]
If I change `currentValue` to `currentDigest` then I get this error:

```Regex Managers must contain currentValueTemplate configuration or regex group named currentValue```

**Reply by viceice** [reply]
you can try to set latest as current value template

**Reply by rarkins** [reply, marked as answer]
as in `"currentValueTemplate": "latest"`.

**Reply by juusopiikkila** [reply]
Ah yes that did the trick. Thank you!
